src/main/java/com/google/songgame/servlets/RoomServlet.java

```diff
@@ -98,4 +120,12 @@ private String getUserId(HttpServletRequest request) throws IOException {
 
     return userId;
   }
+
+  private String getUsername(String userId) {
```

Comment:
I'm not sure if we should do this (by this I mean the suggestion I am about to propose).

Instead of fetching the users one by one from datastore, since we already have the userIdList on line 88, we can get all users in one query to datastore and return all records usernames .

Reply:
@alekseychaikovsky Fixed it to get userIds from userList in one query. It is a lot quicker now!